src/Microsoft.Diagnostics.Monitoring/DiagnosticServices.cs

```diff
@@ -27,43 +25,60 @@ public DiagnosticServices(ILogger<DiagnosticsMonitor> logger)
 
         public IEnumerable<int> GetProcesses()
         {
-            //TODO This won't work properly with multi-container scenarios that don't share the process space.
-            //TODO We will need to use DiagnosticsAgent if we are the server.
-            return DiagnosticsClient.GetPublishedProcesses();
+            try
+            {
+                //TODO This won't work properly with multi-container scenarios that don't share the process space.
+                //TODO We will need to use DiagnosticsAgent if we are the server.
+                return DiagnosticsClient.GetPublishedProcesses();
+            }
+            catch (UnauthorizedAccessException)
+            {
+                throw new InvalidOperationException("Unable to enumerate processes.");
+            }
         }
 
-        public async Task<Stream> GetDump(int pid, DumpType mode)
+        public async Task<OperationResult<Stream>> GetDump(int? pid, DumpType mode)
```

Comment:
>OperationResult [](start = 26, length = 15)

This api has changed to return something else in my current code (an interface). Perhaps i can merge this together with your OperationResult.